File: verify_fix.py

from playwright.sync_api import sync_playwright
import os

def run():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()
        page.set_viewport_size({"width": 1280, "height": 720})
        
        file_path = os.path.abspath("index.html")
        page.goto(f"file://{file_path}")
        
        # Trigger modal with correct signature: title, message, choices
        page.evaluate("""
            window.showModal(
                'Test Title', 
                'Comparison Check.', 
                [{ label: 'Action', value: 'action' }]
            );
        """)
        
        # Wait for modal visibility
        page.wait_for_selector("#modal-overlay")
        page.wait_for_timeout(500)
        
        page.screenshot(path="/home/jules/verification/fix_check.png")
        print("Screenshot generated.")
        browser.close()

if __name__ == "__main__":
    run()

File: verify_fix.js

const { chromium } = require('playwright');
const path = require('path');

(async () => {
  const browser = await chromium.launch({
    args: ['--no-sandbox', '--disable-setuid-sandbox']
  });
  const page = await browser.newPage();
  
  // Set viewport to see toolbar and modal
  await page.setViewportSize({ width: 1280, height: 720 });
  
  const filePath = path.resolve(__dirname, 'index.html');
  await page.goto(`file://${filePath}`);
  
  // Locate the save button in the top right to compare
  const saveBtn = page.locator('#saveBtn');
  // Trigger a modal
  await page.evaluate(() => {
    window.showModal('Comparison Check.', {
      title: 'Test Title',
      choices: [{ label: 'Action', value: 'action' }]
    });
  });
  const modal = page.locator('#modal-overlay');
  await expect(modal).toBeVisible();
  
  // Wait a moment for transitions
  await page.waitForTimeout(500);

  await page.screenshot({ path: '/home/jules/verification/fix_check.png' });
  console.log("Screenshot generated.");
  
  await browser.close();
})();

async function expect(locator) {
  const count = await locator.count();
  if (count === 0) throw new Error("Element not found");
  const visible = await locator.isVisible();
  if (!visible) throw new Error("Element not visible");
}
